pkg/modules/authorization.go

```diff
@@ -631,6 +633,543 @@ func AuthorizationServerDeployment(ctx context.Context, isDeleting bool, op util
+	}
+
+	// get component variables
+	replicas := new(int32)
+	sentinels := ""
+	image := ""
+	vaultAddress := ""
+	vaultRole := ""
+	vaultKVEnginePath := ""
+	vaultSkipCertificateValidation := false
+	vaultCertificate := ""
+	vaultPrivateKey := ""
+	vaultCertificateAuthority := ""
+	leaderElection := true
+	for _, component := range authModule.Components {
+		switch component.Name {
+		case AuthProxyServerComponent:
+			tmp := int32(component.StorageServiceReplicas)
+			replicas = &tmp
+			image = component.StorageService
+			leaderElection = component.LeaderElection
+		case AuthRedisComponent:
+			var sentinelValues []string
+			for i := 0; i < component.RedisReplicas; i++ {
+				sentinelValues = append(sentinelValues, fmt.Sprintf("sentinel-%d.sentinel.%s.svc.cluster.local:5000", i, cr.Namespace))
+			}
+			sentinels = strings.Join(sentinelValues, ", ")
+		case AuthVaultComponent:
+			vaultAddress = component.VaultAddress
+			vaultRole = component.VaultRole
+			vaultKVEnginePath = component.KvEnginePath
+			vaultSkipCertificateValidation = component.SkipCertificateValidation
+			vaultCertificate = component.Certificate
+			vaultPrivateKey = component.PrivateKey
+			vaultCertificateAuthority = component.CertificateAuthority
+		default:
+			continue
+		}
+	}
+
+	// define volumes
+	volumes := []corev1.Volume{
+		{
+			Name: "config-volume",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: "karavi-config-secret",
+				},
+			},
+		},
+		{
+			Name: "csm-config-params",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: "csm-config-params",
+					},
+				},
+			},
+		},
+		{
+			Name: "vault-client-certificate",
+			VolumeSource: corev1.VolumeSource{
+				Projected: &corev1.ProjectedVolumeSource{
+					Sources: []corev1.VolumeProjection{{}},
+				},
+			},
+		},
+	}
+
+	// set vault volumes
+	if vaultCertificateAuthority != "" {
+		volumes[2].VolumeSource.Projected.Sources = append(volumes[2].VolumeSource.Projected.Sources, corev1.VolumeProjection{
+			Secret: &corev1.SecretProjection{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: "vault-certificate-authority",
+				},
+			},
+		})
+	}
+
+	if vaultCertificate != "" && vaultPrivateKey != "" {
+		volumes[2].VolumeSource.Projected.Sources = append(volumes[2].VolumeSource.Projected.Sources, corev1.VolumeProjection{
+			Secret: &corev1.SecretProjection{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: "vault-client-certificate",
+				},
+			},
+		})
+	} else {
+		volumes[2].VolumeSource.Projected.Sources = append(volumes[2].VolumeSource.Projected.Sources, corev1.VolumeProjection{
+			Secret: &corev1.SecretProjection{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: "storage-service-selfsigned-tls",
+				},
+			},
+		})
+	}
+
+	// set deployment
+	deployment := appsv1.Deployment{
```

Comment:
I know there are some differences between V1 and V2 deployments, but is there a way to reuse one deployment struct?

Reply:
Refactored to use a common deployment struct that is modified by the callers of the v1 and v2 flow.

Reply:
Manually deployed Authorization to verify change.